real_time/surface_src/java_src/Alice/src/main/java/com/roboclub/robobuggy/nodes/planners/GPSTrackPlannerNode.java

```diff
@@ -0,0 +1,171 @@
+package com.roboclub.robobuggy.nodes.planners;
+
+import java.io.FileInputStream;
+import java.io.IOException;
+import java.io.InputStreamReader;
+import java.util.LinkedList;
+import java.util.List;
+
+import com.orsoncharts.util.json.JSONArray;
+import com.orsoncharts.util.json.JSONObject;
+import com.orsoncharts.util.json.parser.JSONParser;
+import com.orsoncharts.util.json.parser.ParseException;
+import com.roboclub.robobuggy.main.RobobuggyLogicNotification;
+import com.roboclub.robobuggy.main.RobobuggyMessageLevel;
+import com.roboclub.robobuggy.messages.PoseMessage;
+import com.roboclub.robobuggy.nodes.baseNodes.NodeState;
+import com.roboclub.robobuggy.ros.NodeChannel;
+import com.roboclub.robobuggy.ui.LocTuple;
+
+/**
+ * {@link PathPlannerNode} used to plan a path that follows a previously 
+ * recorded path of GPS waypoints, loaded from a log file.
+ * 
+ * @author Zachary Dawson
+ *
+ */
+public final class GPSTrackPlannerNode extends PathPlannerNode {
+
+	private static final double NEXT_THRESHOLD = 0.1;
+	private static final double BRAKE_THRESHOLD = 0.5;
+	
+	private List<LocTuple> waypoints;
+	
+	private double commandedAngle;
+	private boolean commandedBrake;
+	
+	/**
+	 * Construct a new {@link GPSTrackPlannerNode}
+	 * @param channel {@link NodeChannel} on which to broadcast status
+	 *  information about the node
+	 * @param logFilePath {@link String} representing the path to the log file
+	 *  from which the desired GPS path will be extracted.
+	 */
+	public GPSTrackPlannerNode(NodeChannel channel, String logFilePath) {
+		super(channel);
+		
+		waypoints = new LinkedList<>();
+		commandedAngle = 0.0;
+		commandedBrake = false;
+		
+		/* Read in the log file and extract the GPS path */
+		System.out.println("Starting to read GPS path from file");
+		try {
+			JSONParser parser = new JSONParser();
+			InputStreamReader reader = new InputStreamReader(new FileInputStream(logFilePath),"UTF-8");	
+			JSONObject completeLogFile = (JSONObject) parser.parse(reader); 
+			//may take a while, currently working on a solution to read line by line
+
+			JSONArray sensorDataArray = (JSONArray) completeLogFile.get("sensor_data");
+			for(Object senObj : sensorDataArray) {
+				JSONObject sensor = (JSONObject)senObj;
+				String sensorName = (String) sensor.get("name");
+				if(sensorName == null){
+					new RobobuggyLogicNotification("sensor name is not in this lines log line, this log cannot be repaid",
+							RobobuggyMessageLevel.EXCEPTION);
+				}else{
+					JSONObject sensorParams = (JSONObject) sensor.get("params");
+					if(NodeChannel.getNodeForName(sensorName).equals(NodeChannel.GPS)) {
+						double latitude = (double) sensorParams.get("latitude");
+						double longitude = (double) sensorParams.get("longitude");
+						String latDir = (String) sensorParams.get("lat_direction");
+						String longDir = (String) sensorParams.get("long_direction");
+						boolean north = latDir.equals("N");
```

Comment:
![Codacy](https://www.codacy.com/versioned/images/favicon.png) Issue found: [Position literals first in String comparisons](https://www.codacy.com/app/Mr-Trevor-Decker/RoboBuggy/file/1935738100/issues/source?bid=2938665&fileBranchId=3103406#l73)